e2e/app/page/notebook-preview-page.ts

```diff
@@ -40,9 +39,25 @@ export default class NotebookPreviewPage extends AuthenticatedPage {
     return notebookPage;
   }
 
-  async getFormattedCode(): Promise<string> {
-    const codeContent = await this.waitForCssSelector('#notebook-container pre');
-    return getPropValue<string>(codeContent, 'textContent');
+  getEditLink(): Link {
+    return new Link(this.page, Selector.editLink);
+  }
+
+  getRunPlaygroundModeLink(): Link {
+    return new Link(this.page, Selector.runPlaygroundModeLink);
+  }
+
+  async getFormattedCode(): Promise<string[]> {
```

Comment:
Function returns array instead of a glob of string for more precise string check.